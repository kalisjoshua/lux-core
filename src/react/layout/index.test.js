import React from 'react';
import renderer from 'react-test-renderer';

import registry from '../componentRegistry';

import './index';
const Layout = registry('Layout');

describe('Layout (supplied Layout)', function () {
  it('should log an error because no content/data is provided', function () {
    spyOn(console, 'error');//.and.callThrough();
    console.error.calls.reset();

    expect(console.error.calls.count()).toBe(0);

    renderer.create(<Layout />);
    expect(console.error.calls.count()).toBe(2);
  });

  it('should render an Error page', function () {
    const response = {
      data: {
        data: {},
        error: new Error('Nothing worked.'),
      },
      error: true,
      status: 500,
    };

    const component = renderer.create(
      <Layout {...response}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
