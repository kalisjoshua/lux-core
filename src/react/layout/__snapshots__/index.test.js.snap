exports[`Layout (supplied Layout) should render an Error page 1`] = `
<div
  className="page">
  <div
    id="bannerSkip">
    <p>
      <a
        className="link--plain"
        href="#maincontent"
        title="Skip to main content">
        attrs.title
      </a>
    </p>
  </div>
  <header
    className="main-header"
    role="banner">
    <h1>
      <a
        className="link--heroOrSomething"
        href="/">
        Home
      </a>
    </h1>
    <navigation
      id="mainnavigation"
      role="navigation">
      <ul>
        <li
          className="">
          <a
            className=""
            href="/"
            onClick={[Function]}
            title="Home">
            Home
          </a>
        </li>
      </ul>
    </navigation>
    <noscript />
  </header>
  <main
    className="body-divider pbl"
    id="maincontent"
    role="main" />
  <footer
    role="contentinfo">
    <p>
      Pages generated by
      <a
        className="link--plain"
        href="https://GitHub.com/luxui">
        Luxui
      </a>
      ENGINEERED to Amaze, by
      <a
        className="link--plain"
        href="http://quickenloans.com">
        Quicken Loans
      </a>
    </p>
  </footer>
</div>
`;
